helix-term/src/commands.rs

```diff
@@ -4083,12 +4083,15 @@ fn match_brackets(cx: &mut Context) {
     let (view, doc) = current!(cx.editor);
 
     if let Some(syntax) = doc.syntax() {
-        let text = doc.text().slice(..);
+        let text = doc.text();
+        let text_slice = text.slice(..);
+        let is_select = cx.editor.mode == Mode::Select;
         let selection = doc.selection(view.id).clone().transform(|range| {
-            if let Some(pos) =
-                match_brackets::find_matching_bracket_fuzzy(syntax, doc.text(), range.cursor(text))
+            let pos = range.cursor(text_slice);
+            if let Some(pos) = match_brackets::find_matching_bracket_fuzzy(syntax, text, pos)
+                .or_else(|| match_brackets::find_matching_bracket_current_line_plaintext(text, pos))
```

Comment:
This doesn't seem to be quite what we want. This always perform the plaintext matching if the TS matching doesn't find a match. Meanwhile for plaintext files this won't work as in that case `doc.syntax()` is `None` (condition above).

I think we can do better for treesitter than the current implementation (which gets easily broken by comments for example). That solution might partially reuse your solution but would be constrained to not leave the current TS node. You can decide whether you want to do that in this PR or a followup PR, that would be the right fix for https://github.com/helix-editor/helix/issues/3357

As it's implemented right now I think it's better to only us the plaintext matching for files without a TS grammar (so `doc.syntax()` would be `None`)

Reply:
I see, this fallback will never run when there's no grammar. However, most of the users of this editor will be coding, and likely have an active TS grammar. My hope is that adding this fallback now already brings improvement, and eventually it will be useless if the TS logic improves.

Reply:
That should settle it: https://github.com/helix-editor/helix/pull/4288/commits/adf0fbd3c8a663ceeaba3050f955c4bdec242381